src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -250,6 +287,13 @@ static ucs_status_t uct_cuda_copy_mem_free(uct_md_h md, uct_mem_h memh)
 
 static void uct_cuda_copy_md_close(uct_md_h uct_md) {
     uct_cuda_copy_md_t *md = ucs_derived_of(uct_md, uct_cuda_copy_md_t);
+    CUdevice dev;
+
+    if (md->cuda_ctx_retained) {
```

Comment:
Maybe simplify to: `if (A && B)` ?
(Relying on `&&` order of execution is a definition in the C standard..)

Reply:
Done.